### ISSUE TITLE
Sections that begin with "tags" are generated twice and empty

### ISSUE THREAD

**Issue by velipso** [gohugoio/hugo#10587]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.109.0 darwin/amd64 BuildDate=unknown
</pre>

### Does this issue reproduce with the latest release?

Yes, I am using the latest release.

---

This logic is wrong:

https://github.com/gohugoio/hugo/blob/f2946da9e806c2bafbdd26707fe339db79bd980b/hugolib/content_map.go#L362

Specifically, if a folder inside `/content` has a prefix matching a taxonomy, then it will be rendered twice, and be buggy.

You can reproduce this with the following commands.

The first build is correct, but the second build shows the buggy behavior.

Notice that the second build will render `/tagstest/index.html` _twice_ -- the first time as page "Tagstest", the second time as "My test".

```bash
# quickstart
hugo new site quickstart
cd quickstart
git init
git submodule add https://github.com/theNewDynamic/gohugo-theme-ananke themes/ananke
echo "theme = 'ananke'" >> config.toml
# create dummy section
mkdir content/test
echo -e "---\ntitle: My test\n---" >> content/test/_index.md
echo -e "---\ntitle: Test post\n---\n\nHello, world" >> content/test/post.md
# generate the page
hugo --debug
...
DEBUG 2022/12/27 19:06:51 Render page Test post to "/test/test/index.html"
DEBUG 2022/12/27 19:06:51 Render page My test to "/test/index.html"
DEBUG 2022/12/27 19:06:51 Render page My New Hugo Site to "/index.html"
DEBUG 2022/12/27 19:06:51 Render page Categories to "/categories/index.html"
DEBUG 2022/12/27 19:06:51 Render page Tags to "/tags/index.html"
...
# rename test folder to tagstest and rebuild
mv content/test content/tagstest
hugo --debug
...
DEBUG 2022/12/27 19:07:40 Render page Test post to "/tagstest/test/index.html"
DEBUG 2022/12/27 19:07:40 Render page Tagstests to "/tagstest/index.html"
DEBUG 2022/12/27 19:07:40 Render page My New Hugo Site to "/index.html"
DEBUG 2022/12/27 19:07:40 Render page Tags to "/tags/index.html"
DEBUG 2022/12/27 19:07:40 Render page Categories to "/categories/index.html"
DEBUG 2022/12/27 19:07:40 Render page My test to "/tagstest/index.html"
...
```

This is affecting me because I set my taxonomies to `t: t`:

```
taxonomies:
  t: t
```

Then any section starting with `t` fails to render correctly (in my case, `/content/tutorials/_index.md`).

**Comment by jmooring** [comment]
Related: with `content/tags-xxx/p1.md` ...

> Error building site: missing taxonomy: tags

**Comment by jmooring** [comment]
This is fixed in 0.123.0.
